### DISCUSSION TITLE
Looking for info/docs on using Brotli

### DISCUSSION THREAD

**Discussion by mushu999** [dotnet/yarp#1546] [Q&A] [answered]
Does Yarp support Brotli compression? I also saw where Kerberos/NTLM isn't supported and wondering what to do if the endpoint requires that for authentication. Looking for where I can read up on this.
TIA!

**Comment by samsp-msft** [comment, marked as answer]
For communication between YARP and the calling client, YARP is essentially a piece of ASP.NET core middleware, so other middleware can be used as well, such as https://docs.microsoft.com/en-us/aspnet/core/performance/response-compression?view=aspnetcore-6.0. I think that will support Brotli.

NTLM/Kerberos is not supported by YARP for pass-through because its a connection-based authentication mechanism. YARP will try to re-use connections from the proxy to the destinations, and so you'd end up with requests on connections authenticated by a different user. #1098 has a bit more detail, and #166.

**Reply by samsp-msft** [reply]
Any compression set by the backend destination, will be passed through YARP to the client. Additionally the proxy can be configured using middleware to perform compression as the response is passed through. That may be beneficial if you have a fast link between the proxy and destination, as it can offload the compression work to the proxy process/hardware.
